doc/docker.md

```diff
@@ -1,111 +1,163 @@
-We will have two configurations:
+ - **webserver** npm dev server
+ - **expungeservice** wsgi server (hosting a Flask app)
+ - **db** postgres database
+ - **nginx** proxies /api/ paths to the backend and everything else to the frontend.
 
-- `dev` for development
-- `prod` for production
+Docker volumes sync the project's source code directories with the frontend and backend dev servers, so that the containers don't need to be rebuilt to reflect code changes, only new package dependencies.
 
-`dev` and `prod` containers will run concurrently, with each `webserver` and `database` containers running on distinct ports. This will allow us to deploy both to the production server and run tests w/o impacting the production database.
+Test Environment Containers
+---------------------------
 
+We run a continuous integregation testing pipe using [Travis CI](https://travis-ci.com/). This runs three containers which are managed in the docker-compose.test.yml file.
 
-**Web Server Container**
+ - **webserver** nginx server that contains the frontend static files, built with webpack
+ - **expungeservice** wsgi server (hosting a Flask app)
+ - **db** postgres database
 
-This container will run the Nginx web server, which serves the static frontend pages, and forwards api calls to the backend service.
+Prod Environment Containers
+---------------------------
 
+The app is deployed to Heroku in just two containers, each served from a standalone Heroku app:
 
-**Database Container**
+ - **webserver** The frontend is an nginx container that serves the static files, and proxies api requests to the backend. It runs at https://recordexpungpdx.herokuapp.com/
+ - **expungeservice** The backend is a wsgi server that runs at https://recordexpungpdxapi.herokuapp.com/
 
-This container will run the PostgreSQL database.
+The database runs in a heroku-postgres plugin that is attached to the backend.
 
+Dockerfiles
+-----------
 
-**Service Container**
+Some variations exist between the Dockerfiles used for each enviroment. TODO: elaborate on this a little bit.
 
-This container will run the Python/Flask backend.
 
+Running the Dev containers
+--------------------------
 
-Running the containers
-----------------------
+Launching and managing the local workspace uses a few written in the project's Makefile:
 
-Development of component source code should often be possible without touching Dockerfiles, so here are a few useful commands that might be sufficient for running and updating docker images in the dev environment.
+Launch the docker enviroment with, and then launch the containers with.
+
+```
+make dev_up
+```
+
+Thist first builds the dev docker images if they don't already exist. It runs the docker stack in detached mode, so you can follow the container of logs with:
+
+```
+make dev_logs
+```
+
+While the dev stack is running, each service is accessible in the dev environment at `localhost:<port>`, at the exposed ports defined in the docker-compose.dev.yml file.
```

Comment:
I'm actually thinking of removing the port mappings; I remember Olli had an issue while trying to access the frontend via port 3000. Maybe just mention that the frontend is accessible at `http://localhost` and you can hit the APIs via the `/api` path.

Reply:
I already found it useful to at least expose the database for making a local connection, and if we expose each it can allow interacting with each without relying on nginx. I should def clarify /expand the wording here.